src/components/UnprovisionedSetup.tsx

```diff
@@ -0,0 +1,129 @@
+4. Once the desired instances are selected, we need to create the SM datasource with the accessToken and SM API url in the jsonData so it can be used in subsequent requests to the SM API
+5. Ensure there are datasources for the selected metrics and logs instances (if not, create them)
+6. Store metadata about the metrics and logs instances in the SM datasource
+7. Import all the dashboards (the name of the newly created logs/metrics datasources is required for this)
+8. Send the selected metric and logs instance metadata to the SM API (with the register/save endpoint)
+9. Reload the page and let the routing take control
+*/
+
+export const UnprovisionedSetup: FC<Props> = ({ pluginId, pluginName }) => {
+  const [apiSetup, setApiSetup] = useState<ApiSetupValues | undefined>();
+  const [apiSetupError, setApiSetupError] = useState<string | undefined>();
+  const [tenantInfo, setTenantInfo] = useState<RegistrationInfo | undefined>();
+  const [instanceSelectionError, setInstanceSelectionError] = useState<string | undefined>();
+
+  const onSetupSubmit = async (setupValues: ApiSetupValues) => {
+    // put api host in plugin jsonData
+    try {
+      await updatePluginJsonData(pluginId, { apiHost: apiSetup?.apiHost ?? DEFAULT_API_HOST });
+      const tenantInfo = await initializeTenant(pluginId, setupValues.adminApiToken);
+      setTenantInfo(tenantInfo);
+      setApiSetup(setupValues);
+    } catch (e) {
+      setApiSetupError(e.message);
+    }
+  };
+
+  const onInstanceSelectionSubmit = async (
+    metricsInstanceId: number | undefined,
+    logsInstanceId: number | undefined
+  ) => {
+    if (!metricsInstanceId || !logsInstanceId) {
+      setInstanceSelectionError('A logs and metrics instance must be selected');
+      return;
+    }
+
+    if (!apiSetup || !apiSetup.apiHost || !tenantInfo?.accessToken) {
+      setInstanceSelectionError('Something went wrong setting up Synthetic Monitoring. Please refresh the page.');
+      return;
+    }
+
+    const hostedMetrics = tenantInfo?.instances.find(tenantInstance => tenantInstance.id === metricsInstanceId);
+    const hostedLogs = tenantInfo?.instances.find(tenantInstance => tenantInstance.id === logsInstanceId);
+    if (!hostedMetrics) {
+      setInstanceSelectionError('Missing metrics instance');
+      return;
+    }
+    if (!hostedLogs) {
+      setInstanceSelectionError('Missing logs instance');
+      return;
+    }
+
+    try {
+      const smDatasource = await initializeSMDatasource(apiSetup.apiHost, tenantInfo.accessToken);
+      await getOrCreateMetricAndLokiDatasources({
+        hostedLogs,
+        hostedMetrics,
+        adminApiToken: apiSetup.adminApiToken,
+        smDatasource,
+      });
+
+      await updatePluginJsonData(pluginId, {
+        logs: {
+          grafanaName: `grafanacloud-${hostedLogs.name}`,
+          hostedId: hostedLogs.id,
+        },
+        metrics: {
+          grafanaName: `grafanacloud-${hostedMetrics.name}`,
```

Comment:
**Question:** are these prefixes going to work in an on-prem setup as well?

Reply:
Yeah, it's a little funny because you can run the plugin on prem, but you _have_ to send your data to a cloud instance. So the plugin has to connect to datasources that connect to an instance running in cloud. I name it this way just because that's how cloud names the datasources, so it will be consistent between the two (and make it clear to the user which instance is holding synthmon data)